lib/iris/tests/unit/config/test_Parallel.py

```diff
@@ -0,0 +1,287 @@
+
+        pool = self.pool
+        get = dask.threaded.get
+        self.patch_set_options.assert_called_once_with(pool=pool, get=get)
+
+    def test__five_workers(self):
+        n_workers = 5
+        Parallel(num_workers=n_workers)
+        self.assertEqual(self.patch_Client.call_count, 0)
+        self.patch_ThreadPool.assert_called_once_with(n_workers)
+
+        pool = self.pool
+        get = dask.threaded.get
+        self.patch_set_options.assert_called_once_with(pool=pool, get=get)
+
+    def test__five_workers__contextmgr(self):
+        n_workers = 5
+        options = Parallel()
+        pool = self.pool
+        get = dask.threaded.get
+
+        with options.context(num_workers=n_workers):
+            self.assertEqual(self.patch_Client.call_count, 0)
+            self.patch_ThreadPool.assert_called_with(n_workers)
+
+            self.patch_set_options.assert_called_with(pool=pool, get=get)
+
+        self.patch_ThreadPool.assert_called_with(self.default_num_workers)
+        self.patch_set_options.assert_called_with(pool=pool, get=get)
+
+    def test_threaded(self):
+        scheduler = 'threaded'
+        Parallel(scheduler=scheduler)
+        self.assertEqual(self.patch_Client.call_count, 0)
+        self.patch_ThreadPool.assert_called_once_with(self.default_num_workers)
+
+        pool = self.pool
+        get = dask.threaded.get
+        self.patch_set_options.assert_called_once_with(pool=pool, get=get)
+
+    def test_multiprocessing(self):
+        scheduler = 'multiprocessing'
+        Parallel(scheduler=scheduler)
+        self.assertEqual(self.patch_Client.call_count, 0)
+        self.patch_ThreadPool.assert_called_once_with(self.default_num_workers)
+
+        pool = self.pool
+        get = dask.multiprocessing.get
+        self.patch_set_options.assert_called_once_with(pool=pool, get=get)
+
+    def test_multiprocessing__contextmgr(self):
+        scheduler = 'multiprocessing'
+        options = Parallel()
+        with options.context(scheduler=scheduler):
+            self.assertEqual(self.patch_Client.call_count, 0)
+            self.patch_ThreadPool.assert_called_with(self.default_num_workers)
+
+            pool = self.pool
+            get = dask.multiprocessing.get
+            self.patch_set_options.assert_called_with(pool=pool, get=get)
+
+        default_get = dask.threaded.get
+        self.patch_ThreadPool.assert_called_with(self.default_num_workers)
+        self.patch_set_options.assert_called_with(pool=pool,
+                                                  get=default_get)
+
+    def test_async(self):
+        scheduler = 'async'
+        Parallel(scheduler=scheduler)
+        self.assertEqual(self.patch_Client.call_count, 0)
+        self.assertEqual(self.patch_ThreadPool.call_count, 0)
+
+        pool = self.pool
+        get = dask.async.get_sync
+        self.patch_set_options.assert_called_once_with(pool=None, get=get)
```

Comment:
@dkillick I'm confused, if pool=None in the assert on L.137, why do you need 'pool = self.pool' on L.135?